paypalscript.py

```diff
@@ -1,44 +1,51 @@
 #!/usr/bin/env python3
 import argparse
 import csv
-import os.path
-
-#
-# Parse and set Input and Output Files
-#
-
-parser = argparse.ArgumentParser(description='Takes a PayPal CSV export file and generates a GnuCash formatted CSV import file.  Default input file name is Paypal Export.csv.  Defaults output file name is output.csv')
-parser.add_argument('-i','--input', help='Input file name', nargs="?", default="Paypal Export.csv", type=str)
-parser.add_argument('-o','--output',help='Output file name', nargs="?", default="Output.csv", type=str)
-args = parser.parse_args()
-
-inputFile = args.input
-
-if not os.path.exists(inputFile.strip('\n')):
-    print("Input file does not exist: " + inputFile)
-    quit()
-
-print("Input File:  " + inputFile)    
-
-outputFile = args.output
-print("Output File: " + outputFile)
-
-#
-# Parse input file and create output file
-#
-
-mapping = {"Date": "Date", "Name": "Name", "Amount": "Net", "Description": "Type"}
-
-def strip_row(row):
-    return {k: row[v] for k,v in mapping.items()}
-
-with open(inputFile, 'r', encoding="utf-8-sig") as filein:
-    csv_filein = [strip_row(row) for row in csv.DictReader(filein) if row["Balance Impact"] == "Credit"]
-    with open(outputFile, 'w', newline='') as fileout:
-        writer = csv.DictWriter(fileout, fieldnames=mapping.keys())
-        writer.writeheader()
-        [writer.writerow(row) for row in csv_filein]
-        fileout.close()
-    filein.close()
 
 
+class PayPalConverter:
+    """
+    Parses Paypal csv files into something GNUCash can read
+    """
+    data = []
+    mapping = {'Date': 'Date', 'Name': 'Name', 'Amount': 'Net', 'Description': 'Type'}
+    paypal_header = 'Balance Impact'
+    balance_type = 'Credit'
+
+    def __init__(self, in_file, out_file):
+        self.in_file = in_file
+        self.out_file = out_file
+
+    def _strip_row(self, row):
+        return {k: row[v] for k, v in self.mapping.items()}
+
+    def read(self):
+        try:
+            with open(self.in_file, 'r', encoding='utf-8-sig') as f:
+                self.data = [self._strip_row(row) for row in csv.DictReader(f) if row[self.paypal_header] ==
+                             self.balance_type]
+        except FileNotFoundError:
+            exit(f"File { self.in_file } not found")
+
+    def write(self):
+        with open(self.out_file, 'w', newline='') as f:
+            writer = csv.DictWriter(f, fieldnames=self.mapping.keys())
+            writer.writeheader()
+            writer.writerows(self.data)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Takes a PayPal CSV export file and generates a GnuCash formatted CSV '
```

Comment:
Is there a best practice that would have us put a docstring for this, or can it not receive one because it's not truly a function?